### PR TITLE
 WIP - Api 14963 add nvc pdf

### PR DESCRIPTION
**Pull request by stiehlrod**
## Description of change
 Adds information to the Supplemental Claims (SC) PDF in order to take the 2nd step towards enabling non-veteran claimants (NVC) to fill out a SC on behalf of a veteran.

## Original issue(s)
[API-14963](https://vajira.max.gov/browse/API-14963)

## Things to know about this PR
There isn't a specific spot for NVC address & phone - We will move forward with the assumption that any NVC data would override the veteran data for boxes 9, 10, 11 (Address, email, phone).

To re-state: Address, email, phone are fields that should be filled with the claimant info if NVC, or the Veteran's info if submitted by the Veteran.

### CODE REVIEW

File: modules/appeals_api/app/models/appeals_api/appellant.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def last_name
     end
 
     def ssn
-      auth_headers["X-VA#{header_prefix}-SSN"]
+      auth_headers["X-VA#{header_prefix}-SSN"] || auth_headers['X-VA-SSN']
     end
```
Comment (stiehlrod) [inline modules/appeals_api/app/models/appeals_api/appellant.rb:25]: This accounts for the different name in SC headers.

Comment (ElleeB) [reply modules/appeals_api/app/models/appeals_api/appellant.rb:25]: I don't think we should have differing header keys. @cilestin?

File: modules/appeals_api/app/services/appeals_api/pdf_construction/supplemental_claim/v2/structure.rb [hunk 2]
```diff
@@ -129,9 +144,10 @@ def fill_autosize_fields
             end
             tmp_path
           end
+          # rubocop:enable Metrics/MethodLength
 
           def additional_pages?
-            additional_issues? || additional_evidence_locations? || form_data.long_signature?
+            additional_issues? || additional_evidence_locations? || form_data.long_signature? || form_data.long_name?
```
Comment (ElleeB) [inline modules/appeals_api/app/services/appeals_api/pdf_construction/supplemental_claim/v2/structure.rb:150]: I can't seem to remember - did we decide that if a name is too long we add it to an additional page? Or should this be removed and just keep the check for the long signature?

File: modules/appeals_api/config/schemas/v2/200995.json [hunk 3]
```diff
@@ -112,15 +113,17 @@
...
-                        { "description": "Claimant's timezone. Used for adding the current date to the signature according to the Claimant's local time. If not specified, the current UTC date will be used for the signature (which may be ahead or behind the local date)." },
+                        { "description": "Claimant's timezone. Used for adding the current date to the signature according to the claimant's local time. If not specified, the current UTC date will be used for the signature (which may be ahead or behind the local date)." },
                         { "$ref": "#/definitions/timezone"}
                       ]
                     }
                   },
                   "required": ["claimantType","address", "phone", "email"],
                   "if": { "properties": { "claimantType": { "const": "other" } } },
                   "then": { "required": ["claimantTypeOtherValue"] }
+
```
Comment (ElleeB) [inline modules/appeals_api/config/schemas/v2/200995.json:126]: I think you can remove this space

File: modules/appeals_api/app/models/appeals_api/appellant.rb (hunk 1 above)
Comment (stiehlrod) [reply modules/appeals_api/app/models/appeals_api/appellant.rb:25]: @cilestin ^^^ ?

**Review by cilestin — APPROVED** [review]

**Pull request merged**